### ISSUE TITLE
Rule proposal: no-useless-escape

### ISSUE THREAD

**Issue by BYK** [eslint/eslint#5460]
## When does this rule warn?

The rule warns when certain characters are escaped unnecessarily in string or regex literals.

### Following are considered errors:
```js
var str = "\ab\c";  // Shouldn't escape `a` or `c`

var regex = /\#\!/;  // Shouldn't escape `#` or `!`
```

## Is this rule preventing an error or is it stylistic?

Looks mostly stylistic but I also think it can prevent potential errors.

## Why is this rule a candidate for inclusion instead of creating a custom rule?

I think this is just a core language rule and we have other rules similar to this. I think core-language related rules that can potentially prevent errors or confusion (why did the author escape this character here, is it a mistake or intentional?) should be included in core.

## Are you willing to create the rule yourself?

Yes.

**Comment by mysticatea** [comment]
:+1: I like this idea.
